arrow/src/main/java/org/apache/iceberg/arrow/vectorized/parquet/VectorizedPageIterator.java

```diff
@@ -0,0 +1,469 @@
+    final int actualBatchSize = getActualBatchSize(expectedBatchSize);
+    if (actualBatchSize <= 0) {
+      return 0;
+    }
+    if (eagerDecodeDictionary) {
+      vectorizedDefinitionLevelReader.readBatchOfDictionaryEncodedVarWidth(
+          vector,
+          numValsInVector,
+          actualBatchSize,
+          nullabilityHolder,
+          dictionaryEncodedValuesReader,
+          dictionary);
+    } else {
+      vectorizedDefinitionLevelReader.readBatchVarWidth(
+          vector,
+          numValsInVector,
+          actualBatchSize,
+          nullabilityHolder,
+          plainValuesReader);
+    }
+    triplesRead += actualBatchSize;
+    this.hasNext = triplesRead < triplesCount;
+    return actualBatchSize;
+  }
+
+  /**
+   * Method for reading batches of fixed width binary type (e.g. BYTE[7]). Spark does not support fixed width binary
+   * data type. To work around this limitation, the data is read as fixed width binary from parquet and stored in a
+   * {@link VarBinaryVector} in Arrow.
+   */
+  public int nextBatchFixedWidthBinary(
+      final FieldVector vector, final int expectedBatchSize, final int numValsInVector,
+      final int typeWidth, NullabilityHolder nullabilityHolder) {
+    final int actualBatchSize = getActualBatchSize(expectedBatchSize);
+    if (actualBatchSize <= 0) {
+      return 0;
+    }
+    if (eagerDecodeDictionary) {
+      vectorizedDefinitionLevelReader.readBatchOfDictionaryEncodedFixedWidthBinary(
+          vector,
+          numValsInVector,
+          typeWidth,
+          actualBatchSize,
+          nullabilityHolder,
+          dictionaryEncodedValuesReader,
+          dictionary);
+    } else {
+      vectorizedDefinitionLevelReader.readBatchOfFixedWidthBinary(
+          vector,
+          numValsInVector,
+          typeWidth,
+          actualBatchSize,
+          nullabilityHolder,
+          plainValuesReader);
+    }
+    triplesRead += actualBatchSize;
+    this.hasNext = triplesRead < triplesCount;
+    return actualBatchSize;
+  }
+
+  /**
+   * Method for reading batches of booleans.
+   */
+  public int nextBatchBoolean(
+      final FieldVector vector,
+      final int expectedBatchSize,
+      final int numValsInVector,
+      NullabilityHolder nullabilityHolder) {
+    final int actualBatchSize = getActualBatchSize(expectedBatchSize);
+    if (actualBatchSize <= 0) {
+      return 0;
+    }
+    vectorizedDefinitionLevelReader
+        .readBatchOfBooleans(vector, numValsInVector, actualBatchSize,
+            nullabilityHolder, plainValuesReader);
+    triplesRead += actualBatchSize;
+    this.hasNext = triplesRead < triplesCount;
+    return actualBatchSize;
+  }
+
+  @Override
+  protected void initDataReader(Encoding dataEncoding, ByteBufferInputStream in, int valueCount) {
+    ValuesReader previousReader = plainValuesReader;
+    this.eagerDecodeDictionary = dataEncoding.usesDictionary() && dictionary != null && !allPagesDictEncoded;
```

Comment:
Looks like this will only use eager decoding when not all pages are dictionary encoded.

What about cases where there is no need to keep the dictionary around, like int columns?